partiql-eval/src/main/kotlin/org/partiql/eval/internal/helpers/ValueUtility.kt

```diff
@@ -51,29 +52,42 @@ internal object ValueUtility {
      */
     fun Datum.getText(): String {
         return when (this.type.code()) {
+            PType.VARIANT -> this.lower().getText()
```

Comment:
No need to be changed here, but this is interesting and makes me think Variant needs to be modified slightly so callers aren't required to call `.lower()` even though the same logic will be applied.

Reply:
FWIW, this is pretty similar to the approach I'm developing in partiql-lang-rust.

There's kind of a parallel with `Dynamic` here:
`Dynamic` is a compile-time type that says introspect the type at runtime, because we don't know statically.
`Variant` is a runtime-value-type that says introspect and/or convert  the value first, because it is not yet computable.

By 'lower'ing the Variant value, you allow it to take part in the rest of the already existing PartiQL computation engine. There are probably other approaches that might have some efficiency gains. But this approach is probably the least invasive/chatty.

Reply:
If you look at the PlanTyper, we skip over type checks when we see dynamic (and variant). The downside to this is that variant is a type that exists in our runtime, and we need to introspect the value (by using `lower()`) before passing it to other areas of our engine. We've done a ton of work to make sure that we can safely receive values (via `PType`) in functions and expressions, and `lower()` allows us to continue to do so. Also leverages our cast tables, comparator, etc. without breaking the typing logic.

So, yes, we need to use `lower()` here and at other places before passing it anywhere else.

That being said, if we _know_ that variant will never be encountered due to static type analysis, we can then decide to forgo `lower()`. But right now, we don't have this.